File: src/test/java/color/ColorConverterTest.java
package color;

import cc.moecraft.logger.utils.ColorConverter;

/**
 * 此类由 Hykilpikonna 在 2018/12/16 创建!
 * Created by Hykilpikonna on 2018/12/16!
 * Github: https://github.com/hykilpikonna
 * QQ: admin@moecraft.cc -OR- 871674895
 *
 * @author Hykilpikonna
 */
public class ColorConverterTest
{
    public static void main(String[] args)
    {
        for (int r = 0; r < 256; r += 16)
        {
            for (int g = 0; g < 256; g += 16)
            {
                for (int b = 0; b < 256; b += 16)
                {
                    System.out.print(ColorConverter.rgbTo8Bit(r, g, b) + "|");
                }
                System.out.print(" ");
            }
            System.out.println();
        }
        
        for (int r = 0; r < 256; r += 16)
        {
            for (int g = 0; g < 256; g += 16)
            {
                for (int b = 0; b < 256; b += 16)
                {
                    System.out.print("\033[38;5;" + ColorConverter.rgbTo8Bit(r, g, b) + "m█");
                }
                System.out.print(" ");
            }
            System.out.println();
        }
        
        for (int r = 0; r < 256; r += 16)
        {
            for (int g = 0; g < 256; g += 16)
            {
                for (int b = 0; b < 256; b += 16)
                {
                    System.out.print("\033[38;2;" + r + ";" + g + ";" + b + "m█");
                }
                System.out.print(" ");
            }
            System.out.println();
        }
    }
}

File: src/test/java/color/ColorConverterTest.kt
package color

import org.hydev.logger.coloring.AnsiColorMode.TRUE_COLOR_24BIT
import org.hydev.logger.coloring.AnsiColorMode.XTERM_256_8BIT
import org.hydev.logger.foreground
import org.hydev.logger.to8Bit
import java.awt.Color

/**
 * 此类由 Hykilpikonna 在 2018/12/16 创建!
 * Created by Hykilpikonna on 2018/12/16!
 * Github: https://github.com/hykilpikonna
 * Meow!
 *
 * @author Hykilpikonna
 */
object ColorConverterTest
{
    @JvmStatic
    fun main(args: Array<String>)
    {
        run {
            var r = 0
            while (r < 256)
            {
                var g = 0
                while (g < 256)
                {
                    var b = 0
                    while (b < 256)
                    {
                        print(Color(r, g, b).to8Bit().toString() + "|")
                        b += 16
                    }
                    print(" ")
                    g += 16
                }
                println()
                r += 16
            }
        }
        run {
            var r = 0
            while (r < 256)
            {
                var g = 0
                while (g < 256)
                {
                    var b = 0
                    while (b < 256)
                    {
                        print(Color(r, r, r).foreground(TRUE_COLOR_24BIT) + "█")
                        b += 16
                    }
                    print(" ")
                    g += 16
                }
                println()
                r += 16
            }
        }
        var r = 0
        while (r < 256)
        {
            var g = 0
            while (g < 256)
            {
                var b = 0
                while (b < 256)
                {
                    print(Color(r, r, r).foreground(XTERM_256_8BIT) + "█")
                    b += 16
                }
                print(" ")
                g += 16
            }
            println()
            r += 16
        }
    }
}
